### ISSUE TITLE
Understanding DNS resolution and actual request in code!!

### ISSUE THREAD

**Issue by abinpaul1** [IngoZenz/personaldnsfilter#89]
I am developing an app which requires blocking websites as one of its functionalities. 
I plan to use your codebase as reference for development. However I couldn't understand the flow of the code at a certain point. It would be great if you could point me in the right direction.
Sorry if this is a really bad noob question!!

After going throught the code I summarized this much:
-- 
A dns request is handled by the **DNSFilterService** which uses the **DNSResolver**. In DNSResolver, the response is obtained using the default DNS Provider and then patched using **DNSResponsePatcher**. This patched DNS response is returned.
So basically only DNS requests are handled by this VPN service right? 
How are the other requests handled ie, for the webpages after DNS resolution. Can you point me to the part of code where the logic for handling and forwarding other requests is written?

I am really sorry if this is a bad question. Any help will be really appreciated.

**Comment by IngoZenz** [comment]
It is actually this easy. DNS requests are interferred and eventually patched (filtered).
This is usually sufficient as network request usually start with a DNS request for resolving Host to IP Address...

**Comment by abinpaul1** [comment]
> It is actually this easy. DNS requests are interferred and eventually patched (filtered). This is usually sufficient as network request usually start with a DNS request for resolving Host to IP Address...

Yes..... Its just I couldn't find out where this logic is implmemented. The logic which decides to interfere with only the DNS requests and not other subsequent requests

**Comment by IngoZenz** [comment]
within DNSFilterService.java via Routing the DNS Traffic via the local VPN 

line 481
...addRoute(VIRTUALDNS_IPV4, 32);

line 484 
...addRoute(VIRTUALDNS_IPV6, 128);
